fluent-syntax/src/parser.js

```diff
@@ -693,17 +718,25 @@ export default class FluentParser {
       throw new ParseError('E0014');
     }
 
+    if (ch === '$') {
```

Comment:
I moved the `$` check to the top as this will likely be the most common literal.